src/dune_rules/dune_file.ml

```diff
@@ -288,47 +346,36 @@ module Script = struct
     Path.build generated_dune_file
     |> Io.Untracked.with_lexbuf_from_file ~f:(Dune_lang.Parser.parse ~mode:Many)
     |> List.rev_append from_parent
-    |> parse ~dir ~file:(Some file) ~project
+    |> parse ~mask ~dir ~file:(Some file) ~project
   ;;
 end
 
-let filter_out_stanzas_from_hidden_packages ~visible_pkgs =
-  List.filter_map ~f:(fun stanza ->
-    let include_stanza =
-      match Stanzas.stanza_package stanza with
-      | None -> true
-      | Some package ->
-        let name = Package.name package in
-        Package.Name.Map.mem visible_pkgs name
-    in
-    if include_stanza
-    then Some stanza
-    else (
-      match Stanza.repr stanza with
-      | Library.T l ->
-        Library_redirect.Local.of_private_lib l
-        |> Option.map ~f:Library_redirect.Local.make_stanza
-      | _ -> None))
-;;
-
-let filter_stanzas (mask : Only_packages.t) (dune_files : t list) =
-  match mask with
-  | None -> dune_files
-  | Some visible_pkgs ->
-    List.map dune_files ~f:(fun dune_file ->
-      { dune_file with
-        stanzas = filter_out_stanzas_from_hidden_packages ~visible_pkgs dune_file.stanzas
-      })
+let check_dynamic_stanza =
+  (* CR-rgrinberg: unfortunately this needs to kept in sync with the rules
+     manually *)
+  let err = [ Pp.text "This stanza cannot be generated dynamically" ] in
+  fun stanza ->
+    match Stanza.repr stanza with
+    | Install_conf.T { section = loc, Section Bin; _ } ->
+      User_error.raise ~loc [ Pp.text "binary section cannot be generated dynamically" ]
+    | Coq_stanza.Theory.T { buildable = { Coq_stanza.Buildable.loc; _ }; _ }
+    | Library.T { buildable = { loc; _ }; _ }
+    | Install_conf.T { section = _, Site { loc; _ }; _ }
+    | Executables.T
+        { buildable = { loc; _ }; install_conf = Some { section = _, Section Bin; _ }; _ }
+    | Deprecated_library_name.T { Library_redirect.loc; _ }
+    | Plugin.T { site = loc, (_, _); _ } -> User_error.raise ~loc err
+    | _ -> ()
```

Comment:
Just for my understanding, if one removed this check and used one of the "forbidden" stanzas inside a dynamic include, how would Dune react?

Reply:
It would ignore processing them in some key places. Such as adding them to the library database or the database we use for %{bin:..}.

Reply:
OK, thanks.